app/events/applications/application_deleted_event.rb

```diff
@@ -4,12 +4,10 @@ class Applications::ApplicationDeletedEvent < ApplicationRelatedEvent
 
   # @param [Cinstance] application
   def self.create(application)
-    provider = application.provider_account || Account.new
-
     new(
       application: MissingModel::MissingApplication.new(id: application.id),
       metadata: {
-        provider_id: provider.id,
+        provider_id: application.provider_account_id || application.tenant_id,
```

Comment:
The `provider_account_id` already exists for an Application (Cinstance) because it inherits from Contract
https://github.com/3scale/porta/blob/7a37df095697b436c733684892c9d76042c5fd47/app/models/contract.rb#L42

Reply:
so we don't need `Account.new` ?

Reply:
not at all 😄 before we had:
```ruby
provider = application.provider_account || Account.new
provider_id: provider.id
```
which is equivalent to: `provider_id: application.provider_account&.id` like "If the provider exists, give me its id, otherwise it is `nil`".

Now we have: "give me its provider_id, and if it is not there anymore for whatever reason, give me its tenant_id" 😄

Reply:
Well tenant_id can be `nil` too. It is set by the trigger:

https://github.com/3scale/porta/blob/master/lib/tasks/multitenant/triggers.rake#L450-L452

Reply:
@hallelujah Yes @hallelujah but I can't do magic... if it is not anywhere anymore, it must be nil 😄 but at least this way it will save the cases that it is not nil 😄